File: backend/shared/response.py
"""Consistent API Gateway proxy responses."""

import json
import os
from typing import Any


def api_response(status_code: int, body: Any) -> dict[str, Any]:
    """Build a JSON response with CORS headers for API Gateway."""
    return {
        "statusCode": status_code,
        "headers": {
            "Content-Type": "application/json",
            "Access-Control-Allow-Origin": os.getenv("ALLOWED_ORIGIN", "*"),
            "Access-Control-Allow-Headers": "Content-Type,Authorization",
            "Access-Control-Allow-Methods": "GET,POST,DELETE,OPTIONS",
        },
        "body": json.dumps(body, default=str),
    }


def success(body: Any, status_code: int = 200) -> dict[str, Any]:
    return api_response(status_code, body)


def error(message: str, status_code: int, code: str | None = None) -> dict[str, Any]:
    payload: dict[str, str] = {"message": message}
    if code:
        payload["code"] = code
    return api_response(status_code, payload)


def options_response() -> dict[str, Any]:
    return api_response(204, {})

File: backend/shared/response.js
/**
 * Consistent API Gateway proxy responses.
 */

const { ALLOWED_ORIGIN } = require("../config/environment");

/**
 * Build a JSON response with CORS headers for API Gateway.
 *
 * @param {number} statusCode - HTTP status code.
 * @param {*}      body       - Response body (will be JSON-stringified).
 * @returns {{ statusCode: number, headers: Object, body: string }}
 */
function apiResponse(statusCode, body) {
  return {
    statusCode,
    headers: {
      "Content-Type": "application/json",
      "Access-Control-Allow-Origin": ALLOWED_ORIGIN,
      "Access-Control-Allow-Headers": "Content-Type,Authorization",
      "Access-Control-Allow-Methods": "GET,POST,DELETE,OPTIONS",
    },
    body: JSON.stringify(body),
  };
}

/**
 * Build a success response.
 *
 * @param {*}      body       - Response payload.
 * @param {number} [statusCode=200] - HTTP status code.
 */
function success(body, statusCode = 200) {
  return apiResponse(statusCode, body);
}

/**
 * Build an error response.
 *
 * @param {string}  message    - Human-readable error message.
 * @param {number}  statusCode - HTTP status code.
 * @param {string?} code       - Optional machine-readable error code.
 */
function error(message, statusCode, code) {
  const payload = { message };
  if (code) payload.code = code;
  return apiResponse(statusCode, payload);
}

/**
 * Build an OPTIONS preflight response.
 */
function optionsResponse() {
  return apiResponse(204, {});
}

module.exports = { apiResponse, success, error, optionsResponse };
